### PR TITLE
[Dem] Add static and dynamic friction to DemStructuresCoupling

### PR DESCRIPTION
**Pull request by joaquinirazabal**
**Description**
Add STATIC_FRICTION and DYNAMIC_FRICTION to DemStructureCoupling instead of using simply FRICTION. This was added to the DEMApplication in April.
Bugs in two DEM CLs have also been solved.

**Changelog**
- Add STATIC_FRICTION and DYNAMIC_FRICTION to DemStructureCoupling.
- Bugfix in  DEM CLs

### CODE REVIEW

File: applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py [hunk 1]
```diff
@@ -935,7 +935,6 @@ def CreateFEMPropertiesForDEFEContact(self):
                 max_id_properties = prop.Id
         props = KratosMultiphysics.Properties(max_id_properties + 1)
         self.created_props_id = max_id_properties + 1
-        props[KratosDEM.FRICTION] =  -0.5773502691896257
```
Comment (AlejandroCornejo) [inline applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: Are you sure that I have to remove the FRICTION? I tried and I got an error I think :S

**Review by AlejandroCornejo — CHANGES_REQUESTED** [review]

File: applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py (hunk 1 above)
Comment (joaquinirazabal) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: Mmmmm... Let me check...

Comment (joaquinirazabal) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: @AlejandroCornejo, did you try with this branch?

Comment (maceligueta) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: The idea is to end up removing FRICTION and substitute it by 2 variables (static and dynamic). If it's giving problems, we should check why. We can debug a minimal example if you send it to us @AlejandroCornejo .

Comment (AlejandroCornejo) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: I've tried to remove it and the DemCouplingStructure gives an error when transfering entities

Comment (joaquinirazabal) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: Sorry, @AlejandroCornejo, but I cannot reproduce the error you are saying. In the master branch it should give an error, but in this branch it shouldn't. To solve that is why I change the line in the file applications/DemStructuresCouplingApplication/custom_utilities/dem_structures_coupling_utilities.h

Comment (joaquinirazabal) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: @AlejandroCornejo, I am properly running a simulation with the FemToDemApplication in this branch, If there is any problem, can you send me an example for debugging?

Comment (AlejandroCornejo) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: Hi! Is it merged with the latest master? In thiscase I'll try it now

Comment (joaquinirazabal) [reply applications/FemToDemApplication/python_scripts/MainCouplingFemDem.py:955]: Yes, it is.

**Comment by AlejandroCornejo** [comment]
The FEM-DEM tests are running fine. Go ahead with the merge!!

**Review by AlejandroCornejo — APPROVED** [review]

**Pull request merged**